src/types/poly_func.rs

```diff
@@ -0,0 +1,526 @@
+//! Polymorphic Function Types
+
+use crate::extension::{ExtensionRegistry, SignatureError};
+use itertools::Itertools;
+
+use super::{
+    type_param::{check_type_args, TypeArg, TypeParam},
+    FunctionType, Substitution,
+};
+
+/// A polymorphic function type, e.g. of a [Graph], or perhaps an [OpDef].
+/// (Nodes/operations in the Hugr are not polymorphic.)
+///
+/// [Graph]: crate::values::PrimValue::Function
+/// [OpDef]: crate::extension::OpDef
+#[derive(
+    Clone, PartialEq, Debug, Eq, derive_more::Display, serde::Serialize, serde::Deserialize,
+)]
+#[display(
+    fmt = "forall {}. {}",
```

Comment:
Perhaps we could omit the `forall .` if there are no bound variables; it does look a little odd. But in that case we'd want some outer label (e.g. "PolyFuncType") to differentiate from `FunctionType`, a distinct type in Rust